### ISSUE TITLE
Wish: Color output graph lines by group color.

### ISSUE THREAD

**Issue by L-Vettz** [DynamoDS/Dynamo#9520]
I’ve been working on a relatively complicated graph over the last two days and I find it hard to keep track of what input goes to what output once the graph starts looking less like math and more like spaghetti. I had an idea that I don’t know if I had seen before. Grouping nodes is something almost everybody does to keep things organized with colors and titles. Would it be possible to have the output lines coming out of any given group match the color of the group it is coming out of?

I’m already using node to code and creating custom nodes for repeated processes as much as I can, and honestly its not like the mess is actually stopping me. I just thought it was an idea worth bringing up.

**Comment by mjkkirschner** [comment]
I like it - or something similar. I guess the problem underlying this is that you want to be able to discern which connector is attached to which port more easily right?

**Comment by L-Vettz** [comment]
Yea pretty much. Groups do a great job of splitting a graph into specific chunks so I can identify what parts of the graph feed what parts. For big graphs it would be nice to be able to identify which of those groups is feeding which other groups. So if I have a problem with a group and there are say 8 or 9 inputs going into that group, I can easily distinguish what inputs are coming from which other groups and where I need to troubleshoot if those inputs share the group color (or some other identifier) as their origin groups.


For smaller graphs or nodes within groups themselves its not much of an issue, as its pretty clear what goes where. But when you get to a graph with dozens of nodes that all interconnect it can get hard to follow.

**Comment by Eckart-S** [comment]
You could also try out the IГIS Dynamo Extension that is able to color nodes and connectors:
https://forum.dynamobim.com/t/i-is-extension-dynamo-2-graph-colorization-customization/35227
